### PR TITLE
hooks: create snapd directory skeleton

### PR DESCRIPTION
**Pull request by zyga**
Even when running from snapd snap we sometimes assume that certain
directories exist in the base snap where we implant ourselves. We need
the void directory for sure, because the tests showed it was missing and
it is infeasible for snap-confine to always create it. I suspect
something similar will happen to the GL directories but testing in that
area is partial, at best. The directory structure is modelled after
snapd.mk from the snapd tree's packaging/ directory. The list will be
duplicated now but it's better than not having those directories in the
first place.

An existing hook that was creating "classic directory" was removed, it
was creating the hostfs directory and I think it warrants that we make
all of the directories at once in one hook.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: hooks/700-tree-skeleton.chroot [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh -ex
+# This sequence is modelled after snapd's packaging/snapd.mk it also contains
+# the important "void" directory, along with the special permissions it
+# carries.
+echo "I: Creating snapd skeleton directory structure"
```
Comment (mvo5) [inline hooks/700-tree-skeleton.chroot]: Hey, maybe 700-snapd-tree-skelegon.chroot (or something) as the name?

File: hooks/700-tree-skeleton.chroot [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+install -m 111 -d var/lib/snapd/void
```
Comment (mvo5) [inline hooks/700-tree-skeleton.chroot]: We have a hook-tests/ with corresponding tests for each hook, might be nice to add some basic smoke testing there.

**Review by mvo5 — APPROVED** [review]
Looks good thank you for working on this. two suggestions inline.

**Comment by mvo5** [comment]
If @sil2100 agrees I would like to merge this.

**Review by sil2100 — APPROVED** [review]
Looks good, tests passing.

**Pull request merged**